woocommerce/woocommerce
# Update mobile app onboarding modal to be two steps

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woomobile-private/issues/315

#### Example screenshots


https://github.com/woocommerce/woocommerce/assets/1945542/d6b39eaf-72ed-4c89-9189-3422a13ca149





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

TC1

Prerequisite: the site has the the full Jetpack plugin (WPCOM sites have Jetpack by default), and hasn't completed all onboarding tasks. The connected WPCOM user is not linked to an A8C email (magic links don't work for A8C emails)

- Apply the PR changes to the site's WC plugin
- Go to the site wp-admin
- Go to the WooCommerce homescreen with the onboarding tasks, and tap on Get the free WooCommerce mobile app under Things to do next (wp-admin/admin.php?page=wc-admin&mobileAppModal=true)
- Check that the modal now includes a two steps control
- Check that the step 1 modal includes a QR code that goes to https://woocommerce.com/mobile/?utm_source=wc_onboarding_mobile_task. 
- Click on App is installed button
-  Check that step 2 contains a CTA to send a magic link email
- Tap Send the sign-in link --> an email should be sent, and the modal should show that the email has been sent as in the screenshot

TC2 

Non-Jetpack sites
Prerequisite: the site doesn't have the Jetpack plugin

- Apply the PR changes to the site's WC plugin
- Go to the site wp-admin
- Go to the WooCommerce homescreen with the onboarding tasks, and tap on Get the free WooCommerce mobile app under Things to do next (wp-admin/admin.php?page=wc-admin&mobileAppModal=true) 
- Check that the modal should include a QR code that goes to https://woocommerce.com/mobile/?utm_source=wc_onboarding_mobile_task
- Click on App is installed button
- Check that a new QR is displayed that goes to woocommerce://app-login?siteUrl=[your url]&username=[your user name]



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
